libqtile/utils.py

```diff
@@ -548,3 +548,20 @@ def on_resume(msg):  # type: ignore
     # Value is False when the event is over i.e. the machine has woken up
     if not sleeping:
         hook.fire("resume")
+
+
+def get_ppid(pid: int) -> int | None:
+    """
+    Gets the parent process id (ppid) of a given process (represented by the pid in input)
+    """
+    try:
+        # Buffering is taken by the psutil default
+        with open("/proc/{0}/stat".format(pid), "rb", buffering=32 * 1024) as f:
+            data = f.read()
+    except OSError:  # Very unlikely we are not able to read the /proc stat file
+        logger.warning("Unable to read stat file for pid={0}".format(pid))
```

Comment:
```suggestion
        logger.warning("Unable to read stat file for pid=%s", pid)
```
That should fix your failure. The reason for this format is that the string substitution is only performed if the log level is set at warning or higher.

Reply:
this is a little embarassing, I completly mis-read the pep8 command 👍